indigo/indigo-extras/src/test/scala/indigoextras/ui/InputFieldTests.scala

```diff
@@ -192,8 +201,54 @@ object InputFieldTests extends TestSuite {
           actual ==> expected
         }
 
+        "Updated text emits an event" - {
+
+          "if the key is set" - {
+            val key =
+              BindingKey("test")
+
+            val field =
+              InputField("", assets).giveFocus.withKey(BindingKey("test"))
+
+            val actual =
+              field.update(context).flatMap(_.update(context))
+
+            actual.state.text ==> "ABCABC"
+            actual.globalEvents.head ==> InputFieldChange(key, "ABC")
+            actual.globalEvents(1) ==> InputFieldChange(key, "ABCABC")
```

Comment:
For future reference. I look at this today and though.. does that mean they're in the wrong order? But as it turns out, events are pushed into the queue from head to tail, and the queue is a fifo queue:

https://github.com/PurpleKingdomGames/indigo/blob/009b2f3a3f88f6d2fb0631dd1378fe9ba6372446/indigo/indigo-core/src/main/scala/indigo/gameengine/GameLoop.scala#L70